src/init/init.ts

```diff
@@ -247,6 +253,41 @@ function createSelectPrompt(env: YeomanEnvironment): InquirerQuestion {
 }
 
 /**
+ * Checks and returns proxy settings from gitconfig. This function
+ * assumes that gitconfig will either be located locally (project's config file)
+ * or globally (user home)
+ */
+function getProxyConfig() {
+  //
+  let pathToConfig = './.git/config';
+  let meta = readConfigFile(pathToConfig);
+  // Checking if local
+  if (meta && meta.https && meta.https.proxy) {
+    return meta.https.proxy;
+  }
+  pathToConfig = (process.env.HOME || process.env.USERPROFILE) + '/.gitconfig';
+  meta = readConfigFile(pathToConfig);
+  if (meta && meta.https) {
+    return meta.https.proxy;
+  }
+  return;
+}
+
+/**
+ * This function tries to read and convert config file to JSON object.
+ */
+function readConfigFile(filePath: string) {
+  let meta;
+  try {
+    meta = fs.readFileSync(filePath, 'utf8');
+    meta = ini.parse(meta);
+  } catch (error) {
+  }
```

Comment:
should we do nothing here? if so, add a `// do nothing` comment so that readers know it's intentional